features/FEATURE_BLE/ble/pal/PalSecurityManager.h

```diff
@@ -280,6 +280,16 @@ class SecurityManager : private mbed::NonCopyable<SecurityManager> {
             connection_handle_t connection
         ) = 0;
 
+        /**
+         * Indicate that the MIC verification has failed. This could
+         * be due to an invalid CSRK key.
+         *
+         * @param[in] connection connection handle
+         */
+        virtual void on_invalid_mic(
```

Comment:
From the spec:
"If the server has no stored CSRK upon receiving a signed write command, it
shall ignore the received Data PDU. Note that since the server does not
respond to a Signed Write Command, the higher layer application may or may
not be notified of the ignored request. Although the disconnection may be an
adequate indication to the user that the devices need to be paired, it is
recommended that implementers consider providing a more informative
indication to the user that the devices need to be paired to establish a CSRK"